### PR TITLE
Chris

### LINKED ISSUE #13: Use-Case and Misuse-case 2
For each use-case, derive security requirements using misuse case analysis. 
The misusers should be contextualized in your environment of operation and relevant to the interactions identified above. Use names for mis-users that help the reader understand their motives, resources, attack of choice, and the available access to the system to carry out their attack.
Iterate between use and misuse cases to elaborate additional security functions. Embed the final diagram in your report.
Diagraming Advice: Use proper notation as discussed in class materials. Use cases should focus on software features. Go back and forth between use case and misuse case analysis until specific functional security requirements can be identified that are implemented by the software. For misuse cases, prioritize mitigations that are implemented in the OSS project, instead of the environment. Make sure all misuse cases are addressed by use cases. Arrange the diagram to reduce clutter

### LINKED ISSUE #16: Review OSS Project Documentation
Review OSS project documentation for security-related configuration and installation issues. Summarize your observations.

### LINKED ISSUE #17: Planning and Contribution
Include a link to your team GitHub repository that shows your internal project task assignments and collaborations to finish this task. 
Include a reflection of your teamwork for this assignment. What issues occurred? How did you resolve them? What did you plan to change moving forward?

### PR DESCRIPTION
**Pull request by Chrs987**
Pushing the rough draft of my use-case, mis-use case, and prevention. Will work on diagram in a bit

### CODE REVIEW

File: Requirements for SSE.md [hunk 1]
```diff
@@ -4,15 +4,15 @@
 
 ### Case list
 
-#### TODO: Usecase 1 Link
+#### [Usecase 1](#use-case-3)
```
Comment (jmpinoy) [inline Requirements for SSE.md]: I made these changes to, hopefully github has a merge conflict resolver for us to pick which lines we want the master to have. Besides that everything else looks good.

**Review by jmpinoy — DISMISSED** [review]

File: Requirements for SSE.md (hunk 1 above)
Comment (Chrs987) [reply Requirements for SSE.md]: Awesome sounds good! Im sure if worse case scenarios if they are just HREF links we can just edit them and clean it up tomorrow in the Master branch.

File: Requirements for SSE.md [hunk 2]
```diff
@@ -119,18 +118,45 @@ TODO: Insert Diagram photo
...
 
 ## Documentation Sources
 
-TODO: Insert documents referenced above (if any)
+* [IFTTT](https://ifttt.com/)
```
Comment (jmpinoy) [inline Requirements for SSE.md:148]: Can you add these links too, thanks.
[HA Add-on Communication] (https://developers.home-assistant.io/docs/add-ons/communication)
[HA Add-on Security] (https://developers.home-assistant.io/docs/add-ons/security/)
[HA Community Contribution Guidelines] (https://github.com/hassio-addons/repository/blob/master/CONTRIBUTING.md)

**Review by jmpinoy — DISMISSED** [review]

File: Requirements for SSE.md (hunk 2 above)
Comment (Chrs987) [reply Requirements for SSE.md:148]: done

File: Requirements for SSE.md [hunk 3]
```diff
@@ -37,22 +37,21 @@ TODO: Insert Diagram photo
...
.../) and automated many of his IoT devices until they began their "Pro" subscription service which limited the number of integrations he could use unless he paid IFTTT's monthly subscription fee. Bob recently talked with his neighbor who shared a blog post about [HomeAssistant](https://blog.briancmoses.com/2020/09/ditching-ifttt-for-home-assistant.html) that discussed the advantages of bringing all IoT devices under a single hub. Being a not so savvy tech user Bob began researching, setting up HomeAssistant, and integrating all his IoT devices and hubs. Once Bob has all his IoT devices managed by HomeAssistant he begins creating his own automations similar to what he did on IFTTT. HomeAssistant handles all its automations inside the [Configuration.yaml](https://www.home-assistant.io/docs/configuration/) file which is editable via several official HomeAssistant add-ons: File Editor, Samba, and SSH. Bob enables Samba so he can input all his device information in the Configuration.yaml including his username, API tokens, and password information for his Ring Security System, Blink Camera System, and other IoT device accounts and sets up his automations similar to how he had it in IFTTT.
```
Comment (Jbibianochavez) [inline Requirements for SSE.md]: Everything looks good. Reading through this I saw talking should be spoke. Also, no big deal but maybe break up the first sentence. maybe something like this. --> Bob was a user of a popular application called [IFTTT](https://ifttt.com/) and automated many of his IoT devices. They began offering their "Pro" subscription service which limited the number of integrations he could use unless he paid IFTTT's monthly subscription fee.

File: Requirements for SSE.md [hunk 4]
```diff
@@ -37,22 +37,21 @@ TODO: Insert Diagram photo
...
+To prevent the mis-use case, HomeAssisnt employs several safety measures. To prevent an attacker from gaining access to the Configurations.yaml file, the web interface is protected by a password and multi-factor authentication to prevent bruteforce attacks. In the event that a user will enable the Samba addon, Samba requires password authentication and the attacking machine to be on the same local network. In the event that a malicious actor does gain access to the Configuration.yaml, HomeAssistant documentation provides users the ability to input sensitive data in a separate yaml file known as the secrets.yaml file. This will allow the user to input !secrets <AccountName> in place of the sensitive data and the Configuration.yaml file will pull the associated information based on the <Accountname>. However the secrets.yamls file still stores information in plain text. HomeAssistant also gives the user the ability to store sensitive data in the systems Environmental Variables, storing passwords in an Amazon Web Services (AWS) server via [Credstash](https://github.com/fugue/credstash) or on a [keyring](https://github.com/jaraco/keyring) stored on the Operating System(OS).
```
Comment (Jbibianochavez) [inline Requirements for SSE.md:50]: Since Bob was giving out his main network password maybe he can set up a guest network to prevent this. Maybe this is out of scope. If so, please ignore.

File: Requirements for SSE.md [hunk 5]
```diff
@@ -119,18 +118,47 @@ TODO: Insert Diagram photo
...
+
+Lovelace offers users an [automation editor](https://www.home-assistant.io/getting-started/automation/) which gives users a clean and easy way to build their own automation. Here users can enable [Multi-factor authentication](https://www.home-assistant.io/docs/authentication/multi-factor-auth/) and ad an extra level of security to their account. The automation editor converts the users newly created automation to yaml and saves it to the Configuration.yaml file. To give users a more robust editing experience HA offers several ways to directly edit the configuration.yaml file. In their [advanced configuration](https://www.home-assistant.io/getting-started/configuration/) page they list several ways to edit the configuration file by installing addons.
```
Comment (Jbibianochavez) [inline Requirements for SSE.md]: Small typo: ad should be add

**Review by Jbibianochavez — COMMENTED** [review]
Great work, a couple grammar fixes and it is good to go.

File: Requirements for SSE.md (hunk 4 above)
Comment (Chrs987) [reply Requirements for SSE.md:50]: Yeah, a guest network is not something offered by HA to prevent the misuse case. I also did not see anything about that in the documentation or any sort of "network segmentation" so I'm thinking it is out of scope for this assignment. Maybe something we could contribute to in their documentation at a later point.

File: Requirements for SSE.md (hunk 3 above)
Comment (Chrs987) [reply Requirements for SSE.md]: Fixed, nice catch thank you

**Review by Jbibianochavez — APPROVED** [review]

**Review by jmpinoy — APPROVED** [review]
